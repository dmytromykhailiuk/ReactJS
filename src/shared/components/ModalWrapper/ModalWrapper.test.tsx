import { ModalWrapper } from "./";
import { shallow, mount, ReactWrapper } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";
import ReactDOM from "react-dom";

jest.mock("react-dom");
(ReactDOM.createPortal as jest.Mock).mockImplementation((arg) => arg);

describe("ModalWrapper", () => {
  let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    global.document = {
      getElementById: jest.fn().mockReturnValue({} as any),
      body: {
        style: {
          overflow: "visible",
        }
      }
    } as any;

    wrapper = mount(<ModalWrapper onCloseModal={() => {}} />);
  })

  it("should match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should show header when header was passed", () => { 
    const wrapper = shallow(<ModalWrapper header={"Modal"} onCloseModal={() => {}} />);
    wrapper.setProps({ header: "Modal" });
    expect(wrapper.exists(".modal-wrapper__header")).toBeTruthy();
  })

  it("should style be hidden after component mounting", () => { 
    expect(document.body.style.overflow).toEqual("hidden");
  })

  it("should style be hidden after component unmounting", () => { 
    wrapper.unmount();
    expect(document.body.style.overflow).toEqual("visible");
  })
})
