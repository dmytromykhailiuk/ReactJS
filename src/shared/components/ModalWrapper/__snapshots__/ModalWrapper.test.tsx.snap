// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper should match snapshot 1`] = `
<ModalWrapper
  onCloseModal={[Function]}
>
  <div
    className="modal-wrapper"
  >
    <div
      className="modal-wrapper__content-wrapper"
    >
      <div
        className="modal-wrapper__content"
      >
        <CloseMenuButton
          onCloseButtonClicked={[Function]}
        >
          <div
            className="close-menu-button__wrapper"
          >
            <div
              className="close-menu-button__button"
              onClick={[Function]}
            />
          </div>
        </CloseMenuButton>
        <div
          className="modal-wrapper__body"
        />
      </div>
    </div>
  </div>
</ModalWrapper>
`;
